.golangci.yml

```diff
@@ -0,0 +1,49 @@
+service:
+  golangci-lint-version: 1.21.0 # Keep this in sync with .circleci/config.yml and Jenkinsfile.pr
+
+run:
+  timeout: 5m
+  modules-download-mode: readonly
+
+linters-settings:
+  gofmt:
+    simplify: true
+  govet:
+    check-shadowing: true
+    enable-all: true
+
+linters:
+  disable-all: true
+  enable:
+    # - bodyclose
```

Comment:
We can enable this, if we drop the check at https://github.com/mattermost/mattermost-marketplace/blob/master/internal/api/client.go#L29, which seams reasonable to me.


Other question: Why are making sure to read the full body in https://github.com/mattermost/mattermost-marketplace/blob/master/internal/api/client.go#L30? I wasn't able to find a reason for this.

Reply:
I was under the impression the `nil` check was necessary in some cases. I don't mind the linting improvements, but I'm hesitant to start to force it into awkward situations like this.

The reason to consume the whole body in general is to enable keep-alive. It may not matter a lot in our particular case, but there doesn't seem to be much harm in adopting this pattern as-is from the mattermost server.